contents/blog/ab-testing-guide-for-engineers.md

```diff
@@ -0,0 +1,207 @@
+
+For example, if you're making changes to your signup flow and your required sample size is 1,000 new sign-ups. If your daily number of sign-ups is 100, then you'll need to run your experiment for `1,000 / 100 = 10 days`.
+
+Once again, there are many calculators online to help you determine your duration, and PostHog also includes this calculator when creating a new experiment.
+
+**A good rule of thumb is that your experiment duration should be between one week and one month:
+**
+- One week is a good minimum since users may behave differently on weekends or weekdays.
+- One month is a good maximum since otherwise you may delay shipping important product changes.
+
+## When not to A/B test
+
+There's are costs and trade-offs to running A/B tests, and so it may not be a good idea to always run them. Here a few scenarios in which you may not want to run an A/B test:
+
+- **When you lack of sufficient traffic or users, or have time constraints:** It won't be possible to gather enough data to obtain the statistically significant results required to make good decisions.
+
+- **When you have high implementation costs:** Sometimes it may create a lot of technical debt to support and maintain both control and test variants of your experiment.
+
+- **When you have opportunity costs:** The time and resources it takes to implement and analyze A/B tests may be better spent working on other projects.
+
+- **When there are ethical considerations:** It can be tempting to run A/B tests on things such as bug fixes or reliability improvements. While A/B tests can indeed verify your changes are working as expected, you should remember that your users are *real* people that are relying on your product and so you should not run a test for the sake of it. Proceed with caution when testing such changes!
+
+In the above scenarios, it may be better to rely on qualitative feedback or user research for decision making.
+
+## How to implement a good A/B test
+
+Once you're satisfied that your experiment meets the [criteria of a good A/B test](#what-makes-a-good-ab-test), it's time to implement it in your code. This is typically done by using [feature flags](/feature-flags/features) to randomly assign users to an experiment variant, and then logging when users are exposed to the variant i.e., trigger the code path that checks for the feature flag. This enables you to compare metrics in aggregate between the users in your variants.
+
+It's absolutely **essential** to only log the users in your experiment who are actually impacted by the changes you're testing. Users who aren't affected by your test should be excluded. If you include unaffected users, their unchanged behavior will mix in with your results, thereby watering down the impact and the clarity of your findings.
+
+To do so, ensure that checking your feature flag and logging their exposure is the *absolute* last condition you check in your code: 
+
+```
```

Comment:
Could we do a multi language block here? or would that look dumb?

Reply:
This is more pseudo-code than actual code, so I dont think we need to add a multilanguage block

Reply:
I'm thinking like one tab of multilanguage block says "Incorrect" and the other says "Correct" instead of having one long block.

Reply:
ah, nice idea. I'll try it out

Reply:
Tried it out but didnt render correctly. It rendered as 2 separate code blocks. I dont think the multilangauge compeonet is enable in the blog?